Sources/SPMTestSupport/MockRegistry.swift

```diff
@@ -114,21 +135,33 @@ public class MockRegistry {
         }
     }
 
-    private func listReleases(packageIdentity: PackageIdentity) throws -> HTTPClientResponse {
+    private func getPackageMetadata(packageIdentity: PackageIdentity) throws -> HTTPClientResponse {
         guard let (scope, name) = packageIdentity.scopeAndName else {
             throw StringError("invalid package identity \(packageIdentity)")
         }
 
-        let package = self.packages[packageIdentity] ?? [:]
+        let versions = self.packageVersions[packageIdentity] ?? [:]
         let metadata = RegistryClient.Serialization.PackageMetadata(
-            releases: package.keys.reduce(into: [String: RegistryClient.Serialization.PackageMetadata.Release]()) { partial, item in
+            releases: versions.keys.reduce(into: [String: RegistryClient.Serialization.PackageMetadata.Release]()) { partial, item in
                 partial[item] = .init(url: "\(Self.mockRegistryURL.absoluteString)/\(scope)/\(name)/\(item)")
             }
         )
 
+        /*
+         <https://github.com/mona/LinkedList>; rel="canonical",
+         <ssh://git@github.com:mona/LinkedList.git>; rel="alternate"
+         */
+        let sourceControlURLs = self.packagesSourceControlURLs[packageIdentity]
+        let links = sourceControlURLs?.map { url in
+            "<\(url.absoluteString)>; rel=alternate"
```

Comment:
Does this `absoluteString` result need escaping if it contains characters that have special meanings?

Reply:
this is in the testing infrastructure so should be safe enough